pom.xml

```diff
@@ -45,11 +45,15 @@
     <arrow.install.dir>${arrow.script.dir}/build/arrow_install</arrow.install.dir>
     <arrow_root>/usr/local</arrow_root>
     <build_protobuf>ON</build_protobuf>
+    <project.prefix>spark-sql-columnar</project.prefix>
+    <project.name.prefix>OAP Project Spark Columnar Plugin</project.name.prefix>
+    <spark311.version>3.1.1</spark311.version>
```

Comment:
Good question. I put the initial shim layer for Spark 3.1.1 here because when you need to add a shim for Spark 3.1.2, that is for the reason Spark 3.1.2 is different on at least one api aspect with Spark 3.1.1. At that time you will need to create a shim for both Spark 3.1.1 and Spark 3.1.2.  So Spark 3.1.1 shim will be always needed when you have a need to create a shim for another newer version.

Reply:
Of course, Spark 3.1.1 layer will also act as a template when you need to add a new one.